pkg/rt/manager.go

```diff
@@ -17,13 +17,23 @@ type ServiceIdentifier interface {
 	ServiceIDs() []string
 }
 
+// ProductCache represents the api.ProductCache behavior.
+type ProductCache interface {
+	HasAccess(string) bool
+}
+
 // MetricsProvider is a consumer contract for a subscriber manager. It models
 // the method of the prom.Registry which yields a set of Prometheus metrics for
 // a specific service ID.
 type MetricsProvider interface {
 	MetricsFor(serviceID string) *prom.Metrics
 }
 
+type subscriberKey struct {
+	serviceID string
+	product   string
+}
```

Comment:
If product should be part of the subscriber key like this says, that would mean there would be different subscriber types for each product, e.g. a subscriberKey with serviceID=X and product=foo would represent a different subscriber than a subscriberKey with serviceID = and product=bar. But there's only a single subscriber type, it's parameterized on service ID, and and has methods for both (all) of the product features. That means "enabled products" is a property of that subscriber type, not the key. So this doesn't really work.